mbq/client/contrib/tests/test_permissions.py

```diff
@@ -0,0 +1,143 @@
+        self, person_id: str, org_ref: sut.UUIDType
+    ) -> sut.FetchedPermissionsDoc:
+        result: Dict = {"global": self.data.get("global", [])}
+        result.update({org_ref: self.data.get(org_ref, [])})
+        return result
+
+    def fetch_permissions_for_location(
+        self, person_id: str, location_id: int, location_type: sut.RefType
+    ) -> sut.FetchedPermissionsDoc:
+        result: Dict = {"global": self.data.get("global", [])}
+        key = f"{location_id}:{location_type}"
+        result.update({key: self.data.get(key, [])})
+        return result
+
+
+class PermissionsClientTest(TestCase):
+    def setUp(self):
+        test_data = {
+            "org": ["read:invoices"],
+            "1:company": ["read:orders", "write:orders"],
+            "2:vendor": ["read:team"],
+            "global": ["read:global"],
+        }
+        self.client = sut.PermissionsClient(
+            TestOSCoreClient(test_data), cache_name=None
+        )
+
+    def test_cache_key(self):
+        self.assertEqual(
+            self.client._cache_key("person", "123"), "permissions_client:person:123"
+        )
+        self.assertEqual(
+            self.client._cache_key("person2", "456", "company"),
+            "permissions_client:person2:456:company",
+        )
+
+    def test_global_cache_key(self):
+        self.assertEqual(
+            self.client._global_cache_key("person"), "permissions_client:person:global"
+        )
+
+    def test_cache_read_no_cache(self):
+        # Just testing that this doesn't error
+        self.assertEqual(self.client._cache_read("person", "org"), None)
+
+    def test_cache_read_with_empty_cache(self):
+        cache_mock = Mock()
+        self.client.cache = cache_mock
+
+        cache_mock.get_many.return_value = {}
+
+        self.assertEqual(self.client._cache_read("person", "org"), None)
+
+        cache_mock.get_many.assert_called_once_with(
+            ["permissions_client:person:global", "permissions_client:person:org"]
+        )
+
+    def test_cache_read_with_populated_cache(self):
+        cache_mock = Mock()
+        self.client.cache = cache_mock
+
+        cached_doc = {
+            "permissions_client:person:global": "|",
+            "permissions_client:person:org": "read:invoices|",
+        }
+        cache_mock.get_many.return_value = cached_doc
+
+        self.assertEqual(self.client._cache_read("person", "org"), cached_doc)
+
+        cache_mock.get_many.assert_called_once_with(
+            ["permissions_client:person:global", "permissions_client:person:org"]
+        )
+
+    def test_cache_write_with_no_cache(self):
+        permissions_doc = {"org": ["read:invoices"], "global": []}
+
+        cached_doc = {
+            "permissions_client:person:org": "read:invoices|",
+            "permissions_client:person:global": "|",
+        }
+
+        self.assertEqual(
+            self.client._cache_write("person", permissions_doc), cached_doc
+        )
```

Comment:
what is the expected result of doing a `_cache_write` when no cache exists?

Reply:
yeah i need to split this out into two functions as noted in previous comment